polyhost/handler/browser_url.py

```
@@ -0,0 +1,125 @@
+        extension report wins; otherwise the macOS AppleScript fallback is
+        tried (a no-op off macOS / for Firefox). ``None`` for any non-browser
+        app, so a browser URL can never leak onto a native app's overlay."""
+        if not is_browser_app(app_name):
+            return None
+        url = self._effective_url()
+        if url:
+            return url
+        # No fresh extension report — try the zero-install macOS path.
+        try:
+            return self._macos_lookup(app_name)
+        except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)